mvlearn/semi_supervised/ctregression.py

```diff
@@ -0,0 +1,580 @@
+            have label np.nan.
+
+        Returns
+        -------
+        self : returns an instance of self
+        """
+
+        # check whether Xs contain NaN and both Xs and y
+        # are consistent woth each other
+        check_Xs_y_nan_allowed(
+            Xs, y, multiview=True, enforce_views=self.n_views)
+
+        # Xs contain two views
+        view1 = Xs[0]
+        view2 = Xs[1]
+
+        # Storing the indexes of the unlabelled samples
+        U = [i[0] for i in enumerate(y) if np.isnan(i[1])]
+
+        # Storing the indexes of the labelled samples
+        L = [i[0] for i in enumerate(y) if not np.isnan(i[1])]
+
+        X1 = view1
+        X2 = view2
+
+        # making two true labels for each view
+        # So that we can make changes to it without altering original labels
+        y1 = y
+        y2 = y
+
+        # contains the indexes of labelled sample
+        not_null1 = L
+        not_null2 = L
+
+        # fitting the estimator object on the train data
+        self.estimator1.fit(X1.iloc[not_null1], y1.iloc[not_null1])
+        self.estimator2.fit(X2.iloc[not_null2], y2.iloc[not_null2])
+
+        # declaring a variable which keeps tracks
+        # of the number of iteration performed
+        it = 0
+
+        while it < self.n_iter and U:
+            # print("Number of iteration done {} \
+            # \nSize of unlabelled sample {}".format(it, len(U)))
+            it += 1
+            random.seed(self.random_state)
+
+            # Taking a subsample from unlabelled indexes
+            u = random.sample(U, min(len(U), self.subsample_size))
+
+            # constains a list of [delta, index, predicted_value]
+            # and after inclusion of unlabelled index in the training data
+            # value for each subsampled index
+            delta1 = []
+
+            # contains the index of unlabelled sample which will be
+            # included in the other regressor object for further training
+            to_include1 = []
+
+            for i in u:
+
+                # delta is defines as the differece between the predicted
+                # value of k nearest neighbour before
+                delta = 0
+
+                # list of k nearest neighbour for the unlabelled sample
+                omega = self.find_k_nearest_neighbour(X1, y1, i)
+
+                for j in omega:
+
+                    # prediction value of each neighbour before including
+                    # the unlabelled sample in the training view
+                    before_pred = self.estimator1.predict(
+                        np.expand_dims(X1.iloc[j], axis=0)
+                    )
+
+                    # predicted value of unlabelled sample
+                    pred = self.estimator1.predict(
+                        np.expand_dims(X1.iloc[i], axis=0))
+
+                    # include the predicted value of unlabelled sample
+                    y1.iloc[i] = pred
```

Comment:
isn't y (and also y1 and y2) a numpy array? See above comment about setting to Series or DataFrame